client/sites/components/sites-dataviews/index.tsx

```diff
@@ -173,12 +189,60 @@ const DotcomSitesDataViews = ( {
 		viewType: dataViewsState.type,
 	} );
 
+	const getTracksEventDataForFilter = ( fieldId: FieldId, value: any ): TracksEventData | null => {
+		if ( value === undefined ) {
+			return null;
+		}
+
+		switch ( fieldId ) {
+			case 'status': {
+				const statusGroup = siteStatusGroups.find( ( group ) => group.value === value );
+				return [ 'calypso_sites_dashboard_filter_status', statusGroup?.slug ?? '' ];
+			}
+
+			case 'icon':
+			case 'last-interacted':
+			case 'last-publish':
+			case 'plan':
+			case 'site-title':
+			case 'stats':
+				return null;
+
+			default: {
```

Comment:
I think we’re adding unnecessary complexity by anticipating future needs that may or may not arise. Wouldn’t it make more sense to simply return the status filter-related info or null otherwise?

Reply:
You mean remove the parts that ensure every potential `FieldId` type is handled? The idea is to ensure we consider adding the relevant Tracks event if/when new fields are added. There's no way to ensure every field with a `filterBy` config is handled by `getTracksEventDataForFilter`, so this definitely isn't bulletproof, it's just meant to be a helpful reminder.

I pondered this for a bit, and I take your point that opting for a simpler approach may be better, especially when this isn't bulletproof anyway. I've slimmed down the `getTracksEventDataForFilter` and removed some of the types I added.